### PR TITLE
Testing / Linting + Travis

### PR DESCRIPTION
**Pull request by NoiSek**
Partially satisfies #12

- Adds [coverage](https://coveralls.io/github/LikeMyBread/Saylua) and corresponding badge
- Webpack now polyfills Promise automatically (We cannot do this with fetch because reasons)
- Adds `npm run test`, `npm run lint`, `npm run py-test`, `npm run js-test` commands
- Linting and initial tests integrated into Travis, Mocha by default runs any folder named 'tests' located within a `static-source`  folder.

### Migration notes:
- Run `npm install -g eslint eslint-plugin-babel-register inferno mocha`
- Run `pip install -r requirements-dev.txt`
- Devs should now run `npm run test` prior to pushing in order to satisfy Travis.

#### Pytest

Runs python tests from any file that matches `test_*.py` in the root `tests` folder.

Currently only capable of handling placeholders at the moment. Integrating tests into the Python half of our codebase is somewhat difficult given the need for appengine. Will ~~solve in a future PR~~ make a ticket for someone else to deal with.

#### Mocha

Example from `saylua/static-source/lib/tests/test_BaseModel.js`

```javascript
import * as assert from "assert";
import Component from "inferno-component";

import BaseModel from "../Models/BaseModel.js";

describe('Models', function() {
  describe('BaseModel', function() {
    it('should initialize without error', (done) => {
      let model = new BaseModel();
      assert.ok(model);
      done();
    });
  });
});
```

Output:

Mediocre amount of additional info [here](https://mochajs.org/#asynchronous-code).

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [flake8]
-ignore = E111,E114,E121,E124,E128,E402,E712
-max-line-length = 100
+ignore = E111,E114,E121,E124,E128,E261,E266,E402,E712
```
Comment (tiffz) [inline .flake8]: :D

File: requirements-dev.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Default requirements
```
Comment (tiffz) [inline requirements-dev.txt]: Awesome! Thanks for doing this. :D

File: saylua/modules/explore/static-source/es6/Dungeon/Reducers/CoreReducer.js [hunk 3]
```diff
@@ -134,6 +138,7 @@ export const CoreReducer = (state, action) => {
       var newGameClock = state.gameClock;
       newGameClock = playerMoved ? newGameClock + 1 : newGameClock;
 
+      // HACK: This must be removed ASAP.
```
Comment (tiffz) [inline saylua/modules/explore/static-source/es6/Dungeon/Reducers/CoreReducer.js]: :O

**Review by tiffz — APPROVED** [review]

File: requirements-dev.txt (hunk 2 above)
Comment (LikeMyBread) [inline requirements-dev.txt]: I second this, this is a big help!

File: README.md [hunk 4]
```diff
@@ -98,6 +96,6 @@ Making the following change reduces startup time exponentially, and gets rid of
 
 ## Authors
 
-Mike Bradley  
-Noi Sek  
-Tiffany Zhang  
+Mike Bradley
```
Comment (LikeMyBread) [inline README.md]: By the way, these are alphabetized, but if anyone wants to reorder them I'd have no objection.

**Review by LikeMyBread — APPROVED** [review]
A lot of technical debt paid off here, approved with extreme prejudice.

File: README.md (hunk 4 above)
Comment (NoiSek) [inline README.md]: Well, I'm glad you bring it up. I really think the line should look more like this:

```markdown
## Unified Borg Author

Bradley Sek Tiff Noi Mike Zhang
```

Thx and pls consider.

**Pull request merged**